package.json

```diff
@@ -84,6 +84,7 @@
     "ejs": "^2.5.6",
     "eslint-plugin-eslint-plugin": "^0.7.1",
     "eslint-plugin-node": "^4.2.1",
+    "eslint-plugin-throws-on-load": "^1.0.0",
```

Comment:
I think if you mock a dependency with a function that just means that the module exports a function.. I tried checking the docs of proxyquire and couldn't find any way to do some "initialization" logic before the stubbed exports object.

How about something with a getter, like this:
```diff
diff --git a/tests/lib/config/plugins.js b/tests/lib/config/plugins.js
index 8bbff18..f687343 100644
--- a/tests/lib/config/plugins.js
+++ b/tests/lib/config/plugins.js
@@ -42,6 +42,11 @@ describe("Plugins", () => {
             StubbedPlugins = proxyquire("../../../lib/config/plugins", {
                 "eslint-plugin-example": plugin,
                 "@scope/eslint-plugin-example": scopedPlugin,
+                "eslint-plugin-throws-on-load": {
+                    get rules() {
+                        throw new Error('error thrown while loading this module');
+                    }
+                },
                 "./environments": Environments,
                 "../rules": Rules
             });
```

I ran the tests locally and that seems to work, but it relies on the fact that the rules get registered when the plugin is loaded via `Plugins.load`.